### ISSUE TITLE
Get port/ip from Endpoint

### ISSUE THREAD

**Issue by mrocklin** [Akshay-Venkatesh/ucx-py#111]
My guess is that an Endpoint object has information about the local and remote IP address and port, but that these are hidden within the C layer.  Are there functions that we can call to get this information at the Python layer?

I'm thinking of something like the following:

```python
class Endpoint:

    @property
    def remote_address(self):
        return ucp_py_some_function(self.ep)

    @property
    def local_address(self):
        return ucp_py_some_function(self.ep)
```

cc @Akshay-Venkatesh @TomAugspurger

**Comment by TomAugspurger** [comment]
I vaguely recall trying to expose this at the Python layer, but got stuck. I don't remember the details, but it was probably something trivial for someone who really knows C.

**Comment by Akshay-Venkatesh** [comment]
It should be possible to do this in theory but ucx doesn't provide such a facility today because we don't have access to socket handle obtained from a call such as `accept` but only have an `ucp endpoint` handle. We can add it ourselves outside ucx context maybe through `getsockname` and `getpeername`. 

I'll add the above functionality soon.

**Comment by mrocklin** [comment]
It's not a big deal.  I can work around it.  Just a nice-to-have.

**Comment by jakirkham** [comment]
@madsbk, is this already handled with your rewrite?

**Comment by madsbk** [comment]
> @madsbk, is this already handled with your rewrite?

No, but we could exchange IP/Port information when we initiate communication between two endpoints similar to how we exchange _tag_ information

**Comment by quasiben** [comment]
@madsbk I believe this is still an open issue, correct ?

**Comment by madsbk** [comment]
> @madsbk I believe this is still an open issue, correct ?

Yes, this is useful and fairly easy to implement. Will look at it when I find the time :)

**Comment by quasiben** [comment]
Of course!   there is no pressure at all .  I wanted to make sure I had not missed something
